packages/s3-credentials-endpoint/index.js

```diff
@@ -92,6 +95,112 @@ async function displayS3CredentialInstructions(_req, res) {
   res.send(compiled(process.env));
 }
 
+/**
+ * If DISABLE_S3_CREDENTIALS is not "true", returns undefined, otherwise, send a
+ * boom.ServerUnavailable to the caller, Exiting the request.
+ *
+ * @param {Object} res - express request object
+ * @returns {undefined} - when DISABLE_S3_CREDENTIALS is not 'true'
+ */
+function ensureEndpointEnabled(res) {
+  const disableS3Credentials = process.env.DISABLE_S3_CREDENTIALS;
+
+  if (disableS3Credentials && (disableS3Credentials.toLowerCase() === 'true')) {
+    return res.boom.serverUnavailable('S3 Credentials Endpoint has been disabled');
+  }
+  return undefined;
+}
+
+/**
+ * @returns {bool} whether or not the endpoint is configured to send ACL based
+ * credentials.
+ */
+function configuredForACLCredentials() {
+  if (process.env.CMR_ACL_BASED_CREDENTIALS && process.env.CMR_ACL_BASED_CREDENTIALS.toLowerCase() === 'true') {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Parses a "bucket/key/path" to return an array of ["bucket", "key/path"] where
+ * keypath is "/" if not specified.
+ *
+ * @param {string} bucketKeyPath
+ * @returns {Array<string>} Array [bucket, keypath]
+ */
+function parseBucketKey(bucketKeyPath) {
+  try {
+    const parts = bucketKeyPath.split('/');
+    const bucket = parts.shift();
+    const keypath = parts.join('/');
+    return [bucket, `/${keypath}`];
+  } catch (error) {
+    return [undefined, undefined];
+  }
+}
```

Comment:
Would this function be better if it returned an object of `{bucket, keypath}` rather than an array?

Reply:
I agree, returning an object is better.

Reply:
Done.